_posts/2021-03-05-OpenSSL-hardening-updates.markdown

```diff
@@ -0,0 +1,46 @@
+---
+title: Friday in the Void: OpenSSL and Kernel Hardening
+layout: post
+---
+
+The previously announced [OpenSSL switch](https://voidlinux.org/news/2021/02/OpenSSL.html)
+is now underway. Because OpenSSL is a dependency of a large number of packages,
+the full rebuild process is expected to take several days. Syncing between the
+builders and public repositories has been suspended to ensure that the package
+tree remains consistent. Consequently, no new package updates will appear until
+the switch is complete.
+
+Once updates appear, we recommend that you perform a complete system update to
+simplify the transition. Partial updates are possible, but you will need to
+manually trace all OpenSSL dependants installed on your system and update them
+atomically.
+
+---
+
+Since 2016, the default bootloader configruation in Void Linux has set
+the Linux kernel command-line options `slub_debug=P` and `page_poison=1` to
+provide some level of kernel hardening. Kernel series 5.3 and later offer
+alternative measures `init_on_alloc` and `int_on_free` (see [this kernel
+commit](https://github.com/torvalds/linux/commit/6471384af)).
+
+Void's kernels come with the `init_on_alloc` option enabled by default where
+available (*i.e.*, `linux5.4>=5.4.102`, `linux5.10>=5.10.20` and
+`linux5.11>=5.11.3`). You should usually not disable it, as it has a fairly
+minimal impact on performance (within 1%). The `init_on_free` option is more
+expensive (around 5% on average) and needs to be enabled by hand by passing
+`init_on_free=1` on kernel command line. If you have a good reason to disable
+`init_on_alloc`, you can do that similarly by passing `init_on_alloc=0`.
```

Comment:
I don't think we should be judging people :p 

"If you need to disable ..."

Reply:
i don't see how that is judging people

Reply:
Saying "if you have a good reason" isn't necessary, we have already said that generally speaking one shouldn't disable it. If we want to explain how to disable, it should just be said directly, instead of "requiring" a good reason.

Reply:
It isn't really juding, but I restructed the sentence to shave off a few words.